seata-spring-autoconfigure/seata-spring-autoconfigure-core/src/main/java/io/seata/spring/boot/autoconfigure/provider/SpringBootConfigurationProvider.java

```diff
@@ -61,83 +65,103 @@ public Configuration provide(Configuration originalConfiguration) {
                 if (method.getName().startsWith(INTERCEPT_METHOD_PREFIX) && args.length > 0) {
                     Object result;
                     String rawDataId = (String)args[0];
+                    Class<?> dataType = ReflectionUtil.getWrappedClass(method.getReturnType());
+
+                    // 1. Get config value from the system property
                     result = originalConfiguration.getConfigFromSys(rawDataId);
-                    if (null == result) {
-                        if (args.length == 1) {
-                            result = get(convertDataId(rawDataId));
-                        } else {
-                            result = get(convertDataId(rawDataId), args[1]);
+
+                    if (result == null) {
+                        String dataId = convertDataId(rawDataId);
+
+                        // 2. Get config value from the springboot environment
+                        result = getConfigFromEnvironment(dataId, dataType);
+                        if (result != null) {
+                            return result;
+                        }
+
+                        // 3. Get config defaultValue from the arguments
+                        if (args.length > 1) {
+                            result = args[1];
+
+                            if (result != null) {
+                                // See Configuration#getConfig(String dataId, long timeoutMills)
+                                if (dataType.isAssignableFrom(result.getClass())) {
+                                    return result;
+                                } else {
+                                    result = null;
+                                }
+                            }
+                        }
+
+                        // 4. Get config defaultValue from the property object
+                        try {
+                            result = getDefaultValueFromPropertyObject(dataId);
+                        } catch (Throwable t) {
+                            LOGGER.error("Get config '{}' default value from the property object failed:", dataId, t);
```

Comment:
According to the previous configuration priority, the priority of the value object is greater than the default value priority of the parameter

Reply:
The object is no longer a spring bean, so I think the priority is no longer higher than the arguments.